\mainmatter

```{r snRNA, include=FALSE}
knitr::opts_chunk$set(echo      = TRUE,
                      message   = FALSE, 
                      error     = FALSE,
                      cache     = TRUE,
                      fig.align = 'center')
```

# Single-Cell RNA-seq Technologies


```{r echo=FALSE, top, out.width='150%', fig.show='hold', fig.cap="Explosion of scRNA-seq Throughput [-@Svensson2018] "}
knitr::include_graphics("figures/thoughput.JPG")
```

Technological advances increase the efficiency of single-cell sequencing methods
while the cost decreases day by day. In this chapter, we will briefly go through
some of the single-cell RNA-seq technologies.


## Plate-Based SMART-seq


## MARS-seq


## Droplet Microfluidics (DROP-seq)

```{r echo=FALSE, dropseq1, out.width='140%', fig.show='hold', fig.cap="Extraction and Processing of Single-Cell Transcriptomes by Drop-Seq [-@Macosko2015] "}
knitr::include_graphics("figures/DROP-seq.JPG")
```


DROP-seq enables high throughput profiling of thousands of individual cells in a
short period of time by encapsulating each cell individually with a unique
barcode in a droplet. The process takes place within a microfluidic device.  


```{r echo=FALSE, dropseq2A, out.width='150%', fig.show='hold', fig.cap="Drop-Seq barcoding scheme"}
knitr::include_graphics("figures/DROP-seq2A.JPG")
```


After dissociating the complex tissue sample into individual cells, each cell is
suspended in droplets with microparticle beads which deliver barcoded primers
within a microfluidic device. When cells within the droplet are lysed, the mRNA
of these cells is attached to microparticles.A set of beads are generated with
the reverse transcription of mRNAs into cDNAs. These sets of beads are called
`STAMPs`, "single-cell transcriptomes attached to microparticles". Finally, an
RNA-seq library with about 10,000 single-cell transcriptomes was constructed by
barcode `STAMPs` amplifying.


```{r echo=FALSE, dropseq2B, out.width='150%', fig.show='hold', fig.cap="Sequence of primers on the microparticle"}
knitr::include_graphics("figures/DROP-seq2B.JPG")
```


Each barcoded primer bead contains an unique cell barcode and, an unique
molecular identifiers `(UMI)s` along with the common PCR handle. The common PCR
handle sequences used for PCR amplification after `STAMPs` formation. The cell
barcodes are unique for each droplets, therefore, each microparticles contains
primers with the same cell barcode. `UMIs` are unique barcodes to each
transcript, `UMIs` are counted digitially to count unique number of mRNA
transcripts. And, counting `UMIs`  also removes the uncertainty of PCR
duplicates from transcript copy numbers.


Limitations : 

- One issue of this method is most of the droplets are empty and, very few of the
droplets contains both a cell and microparticle.


## 10X Genomics Chromium

```{r echo=FALSE, Genomicsx10, out.width='150%', fig.show='hold', fig.cap="10x Genomics Chromium scheme [-@Zheng2017]"}
knitr::include_graphics("figures/10XGenomics.jpg")
```


The 10X Genomics method is very similar to DROP-seq and, it is first described
in [-@Zheng2017]. First, the barcoded beads are mixed with the cell suspension.
Then, the mixture is capsulated with the emulsion generated by oil. The
microfluidic device and the barcoded beads of the 10X Genomics are engineered to
increase the efficiency of barcoding cells. Therefore almost 90% of the droplets
contain a cell and a barcoded bead. The main difference between the 10X Genomics
Chromium and DROP-seq is the reverse transcription occurs within the droplets.
After the amplification process, the transcriptomics library is constructed. For
each lanes of a 10X Genomics Chromium chip, approximately 20,000 cells can be
barcoded and sequenced simultaneously.



## Nanowell Platforms


## Sample Indexing (Illumina)


## Single-cell Combinatorial Indexing: sciRNA-seq


