\mainmatter

```{r scRNAprocess, include=FALSE}
knitr::opts_chunk$set(echo      = TRUE,
                      message   = FALSE, 
                      error     = FALSE,
                      cache     = TRUE,
                      fig.align = 'center')
```

# Single-cell RNA-seq Data Processing and Quality Control

In this chapter, we will go through quality control and processing methods for
single-cell data.


## Cell Ranger Pipeline

```{r echo=FALSE, cellRanger, out.width='130%', fig.show='hold', fig.cap="Cell Ranger Pipeline [-@10xGenomics2019]"}
knitr::include_graphics("figures/Cell-Ranger.JPG")
```

In 10X Genomics workflow, which is the main workflow that is commonly used
nowdays, `Cell Ranger pipeline` is used to process Chromium single-cell data to
generating feature-barcode matrices, reading alignment, filtering, clustering
and perform other secondary analysis methodologies. 

`Cell Ranger` identifies each sample, cell and molecule. Identification done by
using sequencing barcodes (eg. Illumina i7 and i5 indices) for samples,
cell-specific barcodes for cells and, UMIs for molecules. The typical `Cell
Ranger` results looks like this:

```{r echo=FALSE, cellRanger2, out.width='50%', fig.show='hold', fig.cap="Cell Ranger Pipeline Results [-@10xGenomics2019]"}
knitr::include_graphics("figures/Cell-Ranger2.JPG")
```

In the previous chapter, we mentioned that some of the droplets may contain no
cells. In the qc summary report of the `Cell Ranger pipeline`, `Cell Ranger`
predicts the number of cells within the inputed data and print the estimation in
"Estimated Number of Cells" section. The "Barcode Rank Plot" indicates the
droplets that contains bead but no cell with the grey color. To determine a
cut-off and remove noisy data, bioinformaticians benefit from this plot. The
"Mapping" section of the report indicates percantage of reads that is aligned
successfully to the genome.

* **HINT :** If the mapping rate is low and, you are working with the pure data
(eg. human data) that may indicate contamination.

* **NOTE :** `Cell Ranger` prints error, if there is a quality issue.

### FASTQ File Format 

High-throughput sequencing reads usually outputs `FASTQ` files. `Cell Ranger`
pipeline produces `FASTQ` files as well. Let's take a look at the `FASTQ` file
format below.

```{r echo=FALSE, fastqfile, out.width='150%', fig.show='hold', fig.cap="Structure of a FASTQ file"}
knitr::include_graphics("figures/fastq-file.JPG")
```

Each `FASTQ` file contains three parts. The first line of the `FASTQ` file that
starts with `@`, contains the read identifier, which indicates the position on
the flowcell. The DNA sequence starts from the second line. And the line comes
before another read identifier contains the per-base sequencing quality score
(also called PHRED score). The formula of the PHRED score is given below where P
is the "Probability of base calling error".

```{r echo=FALSE, PHRED, out.width='50%', fig.show='hold', fig.cap="The formula of the PHRED score"}
knitr::include_graphics("figures/Cell-Ranger3.JPG")
```

### Sequencing Quality : FASTQC tool

Sequence quality can be determined by `FASTQC tool`. `FASTQC tool` provides
simple quality control check on raw sequence data that comes from high
throughput sequencing pipelines. @Anders2010

```{r echo=FALSE, FASTQC, out.width='150%', fig.show='hold', fig.cap="FASTQC Results"}
knitr::include_graphics("figures/FASTQC.JPG")
```

FASTQC tool provides:
* Summary statistics (number of reads, â€¦)
* Sequencing quality summaries
* Sequence biases
* Duplicate reads
* Sequence contamination (adapters, etc.)

### Read trimming

Sequencing data analysis pipelines modify the read sequences produced by a
sequencer. There may be sequence biases due to structural oligonucleotides
(adapters, etc.)  and low-quality sequences. Trimming at reading ends is done as
the first operation in a pipeline to avoid sequence biases.

```{r echo=FALSE, trimming, out.width='150%', fig.show='hold', fig.cap="High-throughput sequencing pipeline scheme"}
knitr::include_graphics("figures/trimming.JPG")
```

There are several softwares present to apply trimming, some of them are:

* CellRanger
* cutadapt
* Trim Galore!
* SeqPurge
* trimmomatic

## Demultiplexing

## Genome and transcriptome references

## Read mapping: STAR


## Gene quantification

## Transcript quantification: KALLISTO

## The Single-Cell data

### Gene count data matrix

### Sparse data representation
 
 
 
## Data normalization

##Dealing with Batch Effects

##Data cleanup (filtering)

##Empty droplets

##Empty droplets
 
## Doublet detection and removal: demuxlet

## Doublet detection and removal: DoubletFinder

## Doublet detection and removal: Scrublet

## Single-cell data issues

 
 
 
 
 
## Data imputation / data smoothing
## MAGIC (Markov affinity-based graph imputation of cells)
