<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>References | Single-Cell Bioinformatics: From Technology to Analysis Methodologies</title>
  <meta name="description" content="References | Single-Cell Bioinformatics: From Technology to Analysis Methodologies" />
  <meta name="generator" content="bookdown 0.36 and GitBook 2.6.7" />

  <meta property="og:title" content="References | Single-Cell Bioinformatics: From Technology to Analysis Methodologies" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="References | Single-Cell Bioinformatics: From Technology to Analysis Methodologies" />
  
  
  

<meta name="author" content="İrem B. GÜNDÜZ" />


<meta name="date" content="2025-05-11" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="single-cell-spatial-transcriptomics.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>



<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Single-Cell Bioinformatics: From Technology to Analysis Methodologies </a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preferance</a></li>
<li class="chapter" data-level="1" data-path="why-single-cell-analysis.html"><a href="why-single-cell-analysis.html"><i class="fa fa-check"></i><b>1</b> Why Single-Cell Analysis?</a></li>
<li class="chapter" data-level="2" data-path="single-cell-rna-seq-technologies.html"><a href="single-cell-rna-seq-technologies.html"><i class="fa fa-check"></i><b>2</b> Single-Cell RNA-seq Technologies</a>
<ul>
<li class="chapter" data-level="2.1" data-path="single-cell-rna-seq-technologies.html"><a href="single-cell-rna-seq-technologies.html#plate-based-smart-seq"><i class="fa fa-check"></i><b>2.1</b> Plate-Based SMART-seq</a></li>
<li class="chapter" data-level="2.2" data-path="single-cell-rna-seq-technologies.html"><a href="single-cell-rna-seq-technologies.html#mars-seq"><i class="fa fa-check"></i><b>2.2</b> MARS-seq</a></li>
<li class="chapter" data-level="2.3" data-path="single-cell-rna-seq-technologies.html"><a href="single-cell-rna-seq-technologies.html#droplet-microfluidics-drop-seq"><i class="fa fa-check"></i><b>2.3</b> Droplet Microfluidics (DROP-seq)</a></li>
<li class="chapter" data-level="2.4" data-path="single-cell-rna-seq-technologies.html"><a href="single-cell-rna-seq-technologies.html#x-genomics-chromium"><i class="fa fa-check"></i><b>2.4</b> 10X Genomics Chromium</a></li>
<li class="chapter" data-level="2.5" data-path="single-cell-rna-seq-technologies.html"><a href="single-cell-rna-seq-technologies.html#nanowell-platforms"><i class="fa fa-check"></i><b>2.5</b> Nanowell Platforms</a></li>
<li class="chapter" data-level="2.6" data-path="single-cell-rna-seq-technologies.html"><a href="single-cell-rna-seq-technologies.html#sample-indexing-illumina"><i class="fa fa-check"></i><b>2.6</b> Sample Indexing (Illumina)</a></li>
<li class="chapter" data-level="2.7" data-path="single-cell-rna-seq-technologies.html"><a href="single-cell-rna-seq-technologies.html#single-cell-combinatorial-indexing-scirna-seq"><i class="fa fa-check"></i><b>2.7</b> Single-cell Combinatorial Indexing: sciRNA-seq</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="single-cell-rna-seq-data-processing-and-quality-control.html"><a href="single-cell-rna-seq-data-processing-and-quality-control.html"><i class="fa fa-check"></i><b>3</b> Single-cell RNA-seq Data Processing and Quality Control</a>
<ul>
<li class="chapter" data-level="3.1" data-path="single-cell-rna-seq-data-processing-and-quality-control.html"><a href="single-cell-rna-seq-data-processing-and-quality-control.html#cell-ranger-pipeline"><i class="fa fa-check"></i><b>3.1</b> Cell Ranger Pipeline</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="single-cell-rna-seq-data-processing-and-quality-control.html"><a href="single-cell-rna-seq-data-processing-and-quality-control.html#fastq-file-format"><i class="fa fa-check"></i><b>3.1.1</b> FASTQ File Format</a></li>
<li class="chapter" data-level="3.1.2" data-path="single-cell-rna-seq-data-processing-and-quality-control.html"><a href="single-cell-rna-seq-data-processing-and-quality-control.html#sequencing-quality-fastqc-tool"><i class="fa fa-check"></i><b>3.1.2</b> Sequencing Quality : FASTQC tool</a></li>
<li class="chapter" data-level="3.1.3" data-path="single-cell-rna-seq-data-processing-and-quality-control.html"><a href="single-cell-rna-seq-data-processing-and-quality-control.html#read-trimming"><i class="fa fa-check"></i><b>3.1.3</b> Read trimming</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="single-cell-rna-seq-data-processing-and-quality-control.html"><a href="single-cell-rna-seq-data-processing-and-quality-control.html#demultiplexing"><i class="fa fa-check"></i><b>3.2</b> Demultiplexing</a></li>
<li class="chapter" data-level="3.3" data-path="single-cell-rna-seq-data-processing-and-quality-control.html"><a href="single-cell-rna-seq-data-processing-and-quality-control.html#star-ultrafast-universal-rna-seq-aligner"><i class="fa fa-check"></i><b>3.3</b> STAR: Ultrafast Universal RNA-seq Aligner</a></li>
<li class="chapter" data-level="3.4" data-path="single-cell-rna-seq-data-processing-and-quality-control.html"><a href="single-cell-rna-seq-data-processing-and-quality-control.html#transcript-quantification-kallisto"><i class="fa fa-check"></i><b>3.4</b> Transcript Quantification: KALLISTO</a></li>
<li class="chapter" data-level="3.5" data-path="single-cell-rna-seq-data-processing-and-quality-control.html"><a href="single-cell-rna-seq-data-processing-and-quality-control.html#the-single-cell-data"><i class="fa fa-check"></i><b>3.5</b> The Single-Cell Data</a></li>
<li class="chapter" data-level="3.6" data-path="single-cell-rna-seq-data-processing-and-quality-control.html"><a href="single-cell-rna-seq-data-processing-and-quality-control.html#data-normalization"><i class="fa fa-check"></i><b>3.6</b> Data normalization</a></li>
<li class="chapter" data-level="3.7" data-path="single-cell-rna-seq-data-processing-and-quality-control.html"><a href="single-cell-rna-seq-data-processing-and-quality-control.html#dealing-with-batch-effects"><i class="fa fa-check"></i><b>3.7</b> Dealing with Batch Effects</a></li>
<li class="chapter" data-level="3.8" data-path="single-cell-rna-seq-data-processing-and-quality-control.html"><a href="single-cell-rna-seq-data-processing-and-quality-control.html#data-cleanup-filtering"><i class="fa fa-check"></i><b>3.8</b> Data cleanup (filtering)</a></li>
<li class="chapter" data-level="3.9" data-path="single-cell-rna-seq-data-processing-and-quality-control.html"><a href="single-cell-rna-seq-data-processing-and-quality-control.html#empty-droplets"><i class="fa fa-check"></i><b>3.9</b> Empty Droplets</a></li>
<li class="chapter" data-level="3.10" data-path="single-cell-rna-seq-data-processing-and-quality-control.html"><a href="single-cell-rna-seq-data-processing-and-quality-control.html#doublet-detection-and-removal-demuxlet"><i class="fa fa-check"></i><b>3.10</b> Doublet detection and removal: <em>demuxlet</em></a></li>
<li class="chapter" data-level="3.11" data-path="single-cell-rna-seq-data-processing-and-quality-control.html"><a href="single-cell-rna-seq-data-processing-and-quality-control.html#doublet-detection-and-removal-doubletfinder"><i class="fa fa-check"></i><b>3.11</b> Doublet detection and removal: <em>DoubletFinder</em></a></li>
<li class="chapter" data-level="3.12" data-path="single-cell-rna-seq-data-processing-and-quality-control.html"><a href="single-cell-rna-seq-data-processing-and-quality-control.html#doublet-detection-and-removal-scrublet"><i class="fa fa-check"></i><b>3.12</b> Doublet detection and removal: <em>Scrublet</em></a></li>
<li class="chapter" data-level="3.13" data-path="single-cell-rna-seq-data-processing-and-quality-control.html"><a href="single-cell-rna-seq-data-processing-and-quality-control.html#issues-of-the-single-cell-data"><i class="fa fa-check"></i><b>3.13</b> Issues of the Single-Cell Data</a></li>
<li class="chapter" data-level="3.14" data-path="single-cell-rna-seq-data-processing-and-quality-control.html"><a href="single-cell-rna-seq-data-processing-and-quality-control.html#data-imputation-data-smoothing"><i class="fa fa-check"></i><b>3.14</b> Data imputation / Data Smoothing</a></li>
<li class="chapter" data-level="3.15" data-path="single-cell-rna-seq-data-processing-and-quality-control.html"><a href="single-cell-rna-seq-data-processing-and-quality-control.html#magic-markov-affinity-based-graph-imputation-of-cells"><i class="fa fa-check"></i><b>3.15</b> MAGIC (Markov Affinity-based Graph Imputation of Cells)</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="methods-used-in-single-cell.html"><a href="methods-used-in-single-cell.html"><i class="fa fa-check"></i><b>4</b> Methods Used in Single-Cell</a></li>
<li class="chapter" data-level="5" data-path="cell-trajectory-inference.html"><a href="cell-trajectory-inference.html"><i class="fa fa-check"></i><b>5</b> Cell Trajectory Inference</a></li>
<li class="chapter" data-level="6" data-path="differential-regulation-and-pathway-analysis-in-single-cell.html"><a href="differential-regulation-and-pathway-analysis-in-single-cell.html"><i class="fa fa-check"></i><b>6</b> Differential Regulation and Pathway Analysis in Single-Cell</a></li>
<li class="chapter" data-level="7" data-path="single-cell-epigenomics.html"><a href="single-cell-epigenomics.html"><i class="fa fa-check"></i><b>7</b> Single-Cell Epigenomics</a></li>
<li class="chapter" data-level="8" data-path="single-cell-atac-seq.html"><a href="single-cell-atac-seq.html"><i class="fa fa-check"></i><b>8</b> Single-Cell ATAC-seq</a></li>
<li class="chapter" data-level="9" data-path="single-cell-data-integration.html"><a href="single-cell-data-integration.html"><i class="fa fa-check"></i><b>9</b> Single-Cell Data Integration</a></li>
<li class="chapter" data-level="10" data-path="single-cell-multiomics.html"><a href="single-cell-multiomics.html"><i class="fa fa-check"></i><b>10</b> Single-Cell Multiomics</a></li>
<li class="chapter" data-level="11" data-path="single-cell-spatial-transcriptomics.html"><a href="single-cell-spatial-transcriptomics.html"><i class="fa fa-check"></i><b>11</b> Single-Cell Spatial Transcriptomics</a>
<ul>
<li class="chapter" data-level="11.1" data-path="single-cell-spatial-transcriptomics.html"><a href="single-cell-spatial-transcriptomics.html#integration-of-single-cell-omics-data"><i class="fa fa-check"></i><b>11.1</b> Integration of Single-Cell Omics Data</a></li>
<li class="chapter" data-level="11.2" data-path="single-cell-spatial-transcriptomics.html"><a href="single-cell-spatial-transcriptomics.html#horizontal-integration"><i class="fa fa-check"></i><b>11.2</b> Horizontal Integration</a></li>
<li class="chapter" data-level="11.3" data-path="single-cell-spatial-transcriptomics.html"><a href="single-cell-spatial-transcriptomics.html#vertical-integration"><i class="fa fa-check"></i><b>11.3</b> Vertical Integration</a>
<ul>
<li class="chapter" data-level="11.3.1" data-path="single-cell-spatial-transcriptomics.html"><a href="single-cell-spatial-transcriptomics.html#mofa-mofa"><i class="fa fa-check"></i><b>11.3.1</b> MOFA / MOFA+</a></li>
<li class="chapter" data-level="11.3.2" data-path="single-cell-spatial-transcriptomics.html"><a href="single-cell-spatial-transcriptomics.html#liger-linked-inference-of-genomic-experimental-relationships"><i class="fa fa-check"></i><b>11.3.2</b> LIGER (Linked Inference of Genomic Experimental Relationships)</a></li>
</ul></li>
<li class="chapter" data-level="11.4" data-path="single-cell-spatial-transcriptomics.html"><a href="single-cell-spatial-transcriptomics.html#weighted-nearest-neighbors-wnn"><i class="fa fa-check"></i><b>11.4</b> Weighted Nearest Neighbors (WNN)</a></li>
<li class="chapter" data-level="11.5" data-path="single-cell-spatial-transcriptomics.html"><a href="single-cell-spatial-transcriptomics.html#bridge-integration"><i class="fa fa-check"></i><b>11.5</b> Bridge Integration</a></li>
<li class="chapter" data-level="11.6" data-path="single-cell-spatial-transcriptomics.html"><a href="single-cell-spatial-transcriptomics.html#deep-learning-approaches-multivi"><i class="fa fa-check"></i><b>11.6</b> Deep Learning Approaches: MultiVI</a></li>
<li class="chapter" data-level="11.7" data-path="single-cell-spatial-transcriptomics.html"><a href="single-cell-spatial-transcriptomics.html#single-cell-dna-sequencing"><i class="fa fa-check"></i><b>11.7</b> Single-Cell DNA sequencing</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Single-Cell Bioinformatics: From Technology to Analysis Methodologies</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="references" class="section level1 unnumbered hasAnchor">
<h1>References<a href="references.html#references" class="anchor-section" aria-label="Anchor link to header"></a></h1>

<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div class="csl-entry">
10xGenomics. 2019. <span>“<span class="nocase">Single-Cell RNA-Seq: An Introductory Overview and Tools for Getting Started. Retrieved 06-08-2019. https://community.10xgenomics.com/t5/10x-Blog/Single-Cell-RNA-Seq-An-Introductory-Overview-and-Tools-for/ba-p/547</span>.”</span> <em>10xGenomics</em>, 5–9. <a href="https://www.10xgenomics.com/blog/single-cell-rna-seq-an-introductory-overview-and-tools-for-getting-started">https://www.10xgenomics.com/blog/single-cell-rna-seq-an-introductory-overview-and-tools-for-getting-started</a>.
</div>
<div class="csl-entry">
Anders, Simon. 2010. <span>“<span class="nocase">Babraham Bioinformatics - FastQC A Quality Control tool for High Throughput Sequence Data</span>.”</span> <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">https://www.bioinformatics.babraham.ac.uk/projects/fastqc/</a>.
</div>
<div class="csl-entry">
Andrews, Tallulah S., and Martin Hemberg. 2018. <span>“<span class="nocase">False signals induced by single-cell imputation</span>.”</span> <em>F1000Research</em> 7: 1740. <a href="https://doi.org/10.12688/f1000research.16613.1">https://doi.org/10.12688/f1000research.16613.1</a>.
</div>
<div class="csl-entry">
Argelaguet, Ricard, Denis Arnol, Dmitry Bredikhin, Yann Deloro, Benedikt Velten, John C. Marioni, and Oliver Stegle. 2020. <span>“MOFA+: A Statistical Framework for Comprehensive Integration of Multi-Modal Single-Cell Data.”</span> <em>Genome Biology</em> 21: 111. <a href="https://doi.org/10.1186/s13059-020-02015-1">https://doi.org/10.1186/s13059-020-02015-1</a>.
</div>
<div class="csl-entry">
Ashuach, Tal, Mariano Gabitto, Michael I Jordan, and Nir Yosef. 2021. <span>“MultiVI: Deep Generative Model for the Integration of Multi-Modal Data.”</span> <em>bioRxiv</em>. <a href="https://doi.org/10.1101/2021.08.20.457057">https://doi.org/10.1101/2021.08.20.457057</a>.
</div>
<div class="csl-entry">
Bray, Nicolas L., Harold Pimentel, Páll Melsted, and Lior Pachter. 2016. <span>“<span class="nocase">Near-optimal probabilistic RNA-seq quantification</span>.”</span> <em>Nature Biotechnology</em> 34 (5): 525–27. <a href="https://doi.org/10.1038/nbt.3519">https://doi.org/10.1038/nbt.3519</a>.
</div>
<div class="csl-entry">
Dijk, David van, Roshan Sharma, Juozas Nainys, Kristina Yim, Pooja Kathail, Ambrose J. Carr, Cassandra Burdziak, et al. 2018. <span>“<span class="nocase">Recovering Gene Interactions from Single-Cell Data Using Data Diffusion</span>.”</span> <em>Cell</em> 174 (3): 716–729.e27. <a href="https://doi.org/10.1016/j.cell.2018.05.061">https://doi.org/10.1016/j.cell.2018.05.061</a>.
</div>
<div class="csl-entry">
Dobin, Alexander, Carrie A. Davis, Felix Schlesinger, Jorg Drenkow, Chris Zaleski, Sonali Jha, Philippe Batut, Mark Chaisson, and Thomas R. Gingeras. 2013. <span>“<span class="nocase">STAR: Ultrafast universal RNA-seq aligner</span>.”</span> <em>Bioinformatics</em> 29 (1): 15–21. <a href="https://doi.org/10.1093/bioinformatics/bts635">https://doi.org/10.1093/bioinformatics/bts635</a>.
</div>
<div class="csl-entry">
Eding, Matt. 2009. <span>“<span>Sparse Matrices</span>.”</span> In, 419–44. <a href="https://doi.org/10.1201/9781439811481.ch20">https://doi.org/10.1201/9781439811481.ch20</a>.
</div>
<div class="csl-entry">
Hafemeister, Christoph, and Rahul Satija. 2019. <span>“<span class="nocase">Normalization and variance stabilization of single-cell RNA-seq data using regularized negative binomial regression</span>.”</span> <em>Genome Biology</em> 20 (1): 1–15. <a href="https://doi.org/10.1186/S13059-019-1874-1/FIGURES/6">https://doi.org/10.1186/S13059-019-1874-1/FIGURES/6</a>.
</div>
<div class="csl-entry">
Hao, Yuhan, Stephanie Hao, Erin Andersen-Nissen, et al. 2022. <span>“Bridge Integration of Single-Cell Multi-Omics Data with Dictionary Learning.”</span> <em>bioRxiv</em>. <a href="https://doi.org/10.1101/2022.02.24.481684">https://doi.org/10.1101/2022.02.24.481684</a>.
</div>
<div class="csl-entry">
Hao, Yuhan, Stephanie Hao, Erin Andersen-Nissen, William M Mauck III, Shiwei Zheng, Andrew Butler, Madison J Lee, et al. 2021. <span>“Integrated Analysis of Multimodal Single-Cell Data.”</span> <em>Cell</em> 184 (13): 3573–3587.e29. <a href="https://doi.org/10.1016/j.cell.2021.04.048">https://doi.org/10.1016/j.cell.2021.04.048</a>.
</div>
<div class="csl-entry">
Kang, Hyun Min, Manasa Subramaniam, Sheila Targ, Mai Nguyen, Lenka Maliskova, Erin McCarthy, Emily Wan, et al. 2018. <span>“Multiplexed Droplet Single-Cell RNA-Sequencing Using Natural Genetic Variation.”</span> <em>Nature Biotechnology</em> 36 (1): 89–94. <a href="https://doi.org/10.1038/nbt.4042">https://doi.org/10.1038/nbt.4042</a>.
</div>
<div class="csl-entry">
Kharchenko, Peter V, Lev Silberstein, and David T Scadden. 2014. <span>“Bayesian Approach to Single-Cell Differential Expression Analysis.”</span> <em>Nature Methods</em> 11: 740–42. <a href="https://doi.org/10.1038/nmeth.2967">https://doi.org/10.1038/nmeth.2967</a>.
</div>
<div class="csl-entry">
Lun, Aaron, Samantha Riesenfeld, Tallulah Andrews, The Dao, Tomas Gomes, and John Marioni. 2019. <span>“EmptyDrops: Distinguishing Cells from Empty Droplets in Droplet-Based Single-Cell RNA Sequencing Data.”</span> <em>Genome Biology</em> 20 (March): 63. <a href="https://doi.org/10.1186/s13059-019-1662-y">https://doi.org/10.1186/s13059-019-1662-y</a>.
</div>
<div class="csl-entry">
Macosko, Evan Z., Anindita Basu, Rahul Satija, James Nemesh, Karthik Shekhar, Melissa Goldman, Itay Tirosh, et al. 2015. <span>“Highly Parallel Genome-Wide Expression Profiling of Individual Cells Using Nanoliter Droplets.”</span> <em>Cell</em> 161 (5): 1202–14. https://doi.org/<a href="https://doi.org/10.1016/j.cell.2015.05.002">https://doi.org/10.1016/j.cell.2015.05.002</a>.
</div>
<div class="csl-entry">
McGinnis, Christopher S, Lauren M Murrow, and Zev J Gartner. 2019. <span>“DoubletFinder: Doublet Detection in Single-Cell RNA Sequencing Data Using Artificial Nearest Neighbors.”</span> <em>Cell Systems</em> 8 (4): 329–337.e4. <a href="https://doi.org/10.1016/j.cels.2019.03.003">https://doi.org/10.1016/j.cels.2019.03.003</a>.
</div>
<div class="csl-entry">
<span>“<span class="nocase">Method of the Year 2013</span>.”</span> 2014. Nature Publishing Group. <a href="https://doi.org/10.1038/nmeth.2801">https://doi.org/10.1038/nmeth.2801</a>.
</div>
<div class="csl-entry">
Mistry, M. Freeman, B. Piper, M. 2017. <span>“<span class="nocase">Alignment with STAR | Introduction to RNA-Seq using high-performance computing</span>.”</span> <a href="https://hbctraining.github.io/Intro-to-rnaseq-hpc-O2/lessons/03_alignment.html">https://hbctraining.github.io/Intro-to-rnaseq-hpc-O2/lessons/03_alignment.html</a>.
</div>
<div class="csl-entry">
Stuart, Tim, Andrew Butler, Paul Hoffman, Christoph Hafemeister, Efthymia Papalexi, William M Mauck III, Yuhan Hao, Marlon Stoeckius, Peter Smibert, and Rahul Satija. 2019. <span>“Comprehensive Integration of Single-Cell Data.”</span> <em>Cell</em> 177 (7): 1888–1902.e21. <a href="https://doi.org/10.1016/j.cell.2019.05.031">https://doi.org/10.1016/j.cell.2019.05.031</a>.
</div>
<div class="csl-entry">
Svensson, Valentine, Roser Vento-Tormo, and Sarah A. Teichmann. 2018. <span>“<span class="nocase">Exponential scaling of single-cell RNA-seq in the past decade</span>.”</span> Nature Publishing Group. <a href="https://doi.org/10.1038/nprot.2017.149">https://doi.org/10.1038/nprot.2017.149</a>.
</div>
<div class="csl-entry">
Welch, Joshua D, Vanja Kozareva, Arthur Ferreira, Caleb Vanderburg, Craig Martin, and Evan Z Macosko. 2019. <span>“Single-Cell Multi-Omic Integration Compares and Contrasts Features of Brain Cell Identity.”</span> <em>Cell</em> 177 (7): 1873–1887.e17. <a href="https://doi.org/10.1016/j.cell.2019.05.006">https://doi.org/10.1016/j.cell.2019.05.006</a>.
</div>
<div class="csl-entry">
Wolock, Samuel L, Ramon Lopez, and Allon M Klein. 2019. <span>“Scrublet: Computational Identification of Cell Doublets in Single-Cell Transcriptomic Data.”</span> <em>Cell Systems</em> 8 (4): 281–291.e9. <a href="https://doi.org/10.1016/j.cels.2018.11.005">https://doi.org/10.1016/j.cels.2018.11.005</a>.
</div>
<div class="csl-entry">
Zhang, Xiannian, Tianqi Li, Feng Liu, Yaqi Chen, Jiacheng Yao, Zeyao Li, Yanyi Huang, and Jianbin Wang. 2018. <span>“Comparative Analysis of Droplet-Based Ultra-High-Throughput Single-Cell RNA-Seq Systems.”</span> <em>Molecular Cell</em> 73 (November). <a href="https://doi.org/10.1016/j.molcel.2018.10.020">https://doi.org/10.1016/j.molcel.2018.10.020</a>.
</div>
<div class="csl-entry">
Zheng, Grace, Jessica Terry, Phillip Belgrader, Paul Ryvkin, Zachary Bent, Ryan Wilson, Solongo Ziraldo, et al. 2017. <span>“Massively Parallel Digital Transcriptional Profiling of Single Cells.”</span> <em>Nature Communications</em> 8 (January): 14049. <a href="https://doi.org/10.1038/ncomms14049">https://doi.org/10.1038/ncomms14049</a>.
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="single-cell-spatial-transcriptomics.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/igunduz/the-book-of-a-cell/edit/master/20-References.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": "https://github.com/igunduz/the-book-of-a-cell/blob/master/20-References.Rmd",
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
