<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 2 Single-Cell RNA-seq Technologies | Single-Cell Bioinformatics: From Technology to Analysis Methodologies</title>
  <meta name="description" content="Chapter 2 Single-Cell RNA-seq Technologies | Single-Cell Bioinformatics: From Technology to Analysis Methodologies" />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 2 Single-Cell RNA-seq Technologies | Single-Cell Bioinformatics: From Technology to Analysis Methodologies" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 2 Single-Cell RNA-seq Technologies | Single-Cell Bioinformatics: From Technology to Analysis Methodologies" />
  
  
  

<meta name="author" content="İrem B. GÜNDÜZ" />


<meta name="date" content="2021-12-31" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="why-single-cell-analysis.html"/>
<link rel="next" href="single-cell-rna-seq-data-processing-and-quality-control.html"/>
<script src="libs/header-attrs-2.11/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>



<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Single-Cell Bioinformatics: From Technology to Analysis Methodologies </a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preferance</a></li>
<li class="chapter" data-level="1" data-path="why-single-cell-analysis.html"><a href="why-single-cell-analysis.html"><i class="fa fa-check"></i><b>1</b> Why Single-Cell Analysis?</a></li>
<li class="chapter" data-level="2" data-path="single-cell-rna-seq-technologies.html"><a href="single-cell-rna-seq-technologies.html"><i class="fa fa-check"></i><b>2</b> Single-Cell RNA-seq Technologies</a>
<ul>
<li class="chapter" data-level="2.1" data-path="single-cell-rna-seq-technologies.html"><a href="single-cell-rna-seq-technologies.html#plate-based-smart-seq"><i class="fa fa-check"></i><b>2.1</b> Plate-Based SMART-seq</a></li>
<li class="chapter" data-level="2.2" data-path="single-cell-rna-seq-technologies.html"><a href="single-cell-rna-seq-technologies.html#mars-seq"><i class="fa fa-check"></i><b>2.2</b> MARS-seq</a></li>
<li class="chapter" data-level="2.3" data-path="single-cell-rna-seq-technologies.html"><a href="single-cell-rna-seq-technologies.html#droplet-microfluidics-drop-seq"><i class="fa fa-check"></i><b>2.3</b> Droplet Microfluidics (DROP-seq)</a></li>
<li class="chapter" data-level="2.4" data-path="single-cell-rna-seq-technologies.html"><a href="single-cell-rna-seq-technologies.html#x-genomics-chromium"><i class="fa fa-check"></i><b>2.4</b> 10X Genomics Chromium</a></li>
<li class="chapter" data-level="2.5" data-path="single-cell-rna-seq-technologies.html"><a href="single-cell-rna-seq-technologies.html#nanowell-platforms"><i class="fa fa-check"></i><b>2.5</b> Nanowell Platforms</a></li>
<li class="chapter" data-level="2.6" data-path="single-cell-rna-seq-technologies.html"><a href="single-cell-rna-seq-technologies.html#sample-indexing-illumina"><i class="fa fa-check"></i><b>2.6</b> Sample Indexing (Illumina)</a></li>
<li class="chapter" data-level="2.7" data-path="single-cell-rna-seq-technologies.html"><a href="single-cell-rna-seq-technologies.html#single-cell-combinatorial-indexing-scirna-seq"><i class="fa fa-check"></i><b>2.7</b> Single-cell Combinatorial Indexing: sciRNA-seq</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="single-cell-rna-seq-data-processing-and-quality-control.html"><a href="single-cell-rna-seq-data-processing-and-quality-control.html"><i class="fa fa-check"></i><b>3</b> Single-cell RNA-seq Data Processing and Quality Control</a>
<ul>
<li class="chapter" data-level="3.1" data-path="single-cell-rna-seq-data-processing-and-quality-control.html"><a href="single-cell-rna-seq-data-processing-and-quality-control.html#cell-ranger-pipeline"><i class="fa fa-check"></i><b>3.1</b> Cell Ranger Pipeline</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="single-cell-rna-seq-data-processing-and-quality-control.html"><a href="single-cell-rna-seq-data-processing-and-quality-control.html#fastq-file-format"><i class="fa fa-check"></i><b>3.1.1</b> FASTQ File Format</a></li>
<li class="chapter" data-level="3.1.2" data-path="single-cell-rna-seq-data-processing-and-quality-control.html"><a href="single-cell-rna-seq-data-processing-and-quality-control.html#sequencing-quality-fastqc-tool"><i class="fa fa-check"></i><b>3.1.2</b> Sequencing Quality : FASTQC tool</a></li>
<li class="chapter" data-level="3.1.3" data-path="single-cell-rna-seq-data-processing-and-quality-control.html"><a href="single-cell-rna-seq-data-processing-and-quality-control.html#read-trimming"><i class="fa fa-check"></i><b>3.1.3</b> Read trimming</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="single-cell-rna-seq-data-processing-and-quality-control.html"><a href="single-cell-rna-seq-data-processing-and-quality-control.html#demultiplexing"><i class="fa fa-check"></i><b>3.2</b> Demultiplexing</a></li>
<li class="chapter" data-level="3.3" data-path="single-cell-rna-seq-data-processing-and-quality-control.html"><a href="single-cell-rna-seq-data-processing-and-quality-control.html#genome-and-transcriptome-references"><i class="fa fa-check"></i><b>3.3</b> Genome and transcriptome references</a></li>
<li class="chapter" data-level="3.4" data-path="single-cell-rna-seq-data-processing-and-quality-control.html"><a href="single-cell-rna-seq-data-processing-and-quality-control.html#read-mapping-star"><i class="fa fa-check"></i><b>3.4</b> Read mapping: STAR</a></li>
<li class="chapter" data-level="3.5" data-path="single-cell-rna-seq-data-processing-and-quality-control.html"><a href="single-cell-rna-seq-data-processing-and-quality-control.html#gene-quantification"><i class="fa fa-check"></i><b>3.5</b> Gene quantification</a></li>
<li class="chapter" data-level="3.6" data-path="single-cell-rna-seq-data-processing-and-quality-control.html"><a href="single-cell-rna-seq-data-processing-and-quality-control.html#transcript-quantification-kallisto"><i class="fa fa-check"></i><b>3.6</b> Transcript quantification: KALLISTO</a></li>
<li class="chapter" data-level="3.7" data-path="single-cell-rna-seq-data-processing-and-quality-control.html"><a href="single-cell-rna-seq-data-processing-and-quality-control.html#the-single-cell-data"><i class="fa fa-check"></i><b>3.7</b> The Single-Cell data</a>
<ul>
<li class="chapter" data-level="3.7.1" data-path="single-cell-rna-seq-data-processing-and-quality-control.html"><a href="single-cell-rna-seq-data-processing-and-quality-control.html#gene-count-data-matrix"><i class="fa fa-check"></i><b>3.7.1</b> Gene count data matrix</a></li>
<li class="chapter" data-level="3.7.2" data-path="single-cell-rna-seq-data-processing-and-quality-control.html"><a href="single-cell-rna-seq-data-processing-and-quality-control.html#sparse-data-representation"><i class="fa fa-check"></i><b>3.7.2</b> Sparse data representation</a></li>
</ul></li>
<li class="chapter" data-level="3.8" data-path="single-cell-rna-seq-data-processing-and-quality-control.html"><a href="single-cell-rna-seq-data-processing-and-quality-control.html#data-normalization"><i class="fa fa-check"></i><b>3.8</b> Data normalization</a></li>
<li class="chapter" data-level="3.9" data-path="single-cell-rna-seq-data-processing-and-quality-control.html"><a href="single-cell-rna-seq-data-processing-and-quality-control.html#doublet-detection-and-removal-demuxlet"><i class="fa fa-check"></i><b>3.9</b> Doublet detection and removal: demuxlet</a></li>
<li class="chapter" data-level="3.10" data-path="single-cell-rna-seq-data-processing-and-quality-control.html"><a href="single-cell-rna-seq-data-processing-and-quality-control.html#doublet-detection-and-removal-doubletfinder"><i class="fa fa-check"></i><b>3.10</b> Doublet detection and removal: DoubletFinder</a></li>
<li class="chapter" data-level="3.11" data-path="single-cell-rna-seq-data-processing-and-quality-control.html"><a href="single-cell-rna-seq-data-processing-and-quality-control.html#doublet-detection-and-removal-scrublet"><i class="fa fa-check"></i><b>3.11</b> Doublet detection and removal: Scrublet</a></li>
<li class="chapter" data-level="3.12" data-path="single-cell-rna-seq-data-processing-and-quality-control.html"><a href="single-cell-rna-seq-data-processing-and-quality-control.html#single-cell-data-issues"><i class="fa fa-check"></i><b>3.12</b> Single-cell data issues</a></li>
<li class="chapter" data-level="3.13" data-path="single-cell-rna-seq-data-processing-and-quality-control.html"><a href="single-cell-rna-seq-data-processing-and-quality-control.html#data-imputation-data-smoothing"><i class="fa fa-check"></i><b>3.13</b> Data imputation / data smoothing</a></li>
<li class="chapter" data-level="3.14" data-path="single-cell-rna-seq-data-processing-and-quality-control.html"><a href="single-cell-rna-seq-data-processing-and-quality-control.html#magic-markov-affinity-based-graph-imputation-of-cells"><i class="fa fa-check"></i><b>3.14</b> MAGIC (Markov affinity-based graph imputation of cells)</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="methods-used-in-single-cell.html"><a href="methods-used-in-single-cell.html"><i class="fa fa-check"></i><b>4</b> Methods Used in Single-Cell</a></li>
<li class="chapter" data-level="5" data-path="cell-trajectory-inference.html"><a href="cell-trajectory-inference.html"><i class="fa fa-check"></i><b>5</b> Cell Trajectory Inference</a></li>
<li class="chapter" data-level="6" data-path="differential-regulation-and-pathway-analysis-in-single-cell.html"><a href="differential-regulation-and-pathway-analysis-in-single-cell.html"><i class="fa fa-check"></i><b>6</b> Differential Regulation and Pathway Analysis in Single-Cell</a></li>
<li class="chapter" data-level="7" data-path="single-cell-epigenomics.html"><a href="single-cell-epigenomics.html"><i class="fa fa-check"></i><b>7</b> Single-Cell Epigenomics</a></li>
<li class="chapter" data-level="8" data-path="single-cell-atac-seq.html"><a href="single-cell-atac-seq.html"><i class="fa fa-check"></i><b>8</b> Single-Cell ATAC-seq</a></li>
<li class="chapter" data-level="9" data-path="single-cell-data-integration.html"><a href="single-cell-data-integration.html"><i class="fa fa-check"></i><b>9</b> Single-Cell Data Integration</a></li>
<li class="chapter" data-level="10" data-path="single-cell-multiomics.html"><a href="single-cell-multiomics.html"><i class="fa fa-check"></i><b>10</b> Single-Cell Multiomics</a></li>
<li class="chapter" data-level="11" data-path="single-cell-spatial-transcriptomics.html"><a href="single-cell-spatial-transcriptomics.html"><i class="fa fa-check"></i><b>11</b> Single-Cell Spatial Transcriptomics</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Single-Cell Bioinformatics: From Technology to Analysis Methodologies</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="single-cell-rna-seq-technologies" class="section level1" number="2">
<h1><span class="header-section-number">Chapter 2</span> Single-Cell RNA-seq Technologies</h1>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:top"></span>
<img src="figures/thoughput.JPG" alt="Explosion of scRNA-seq Throughput [-@Svensson2018] " width="150%" />
<p class="caption">
FIGURE 2.1: Explosion of scRNA-seq Throughput <span class="citation">(<a href="#ref-Svensson2018" role="doc-biblioref">2018</a>)</span>
</p>
</div>
<p>Technological advances increase the efficiency of single-cell sequencing methods
while the cost decreases day by day. In this chapter, we will briefly go through
some of the single-cell RNA-seq technologies.</p>
<div id="plate-based-smart-seq" class="section level2" number="2.1">
<h2><span class="header-section-number">2.1</span> Plate-Based SMART-seq</h2>
</div>
<div id="mars-seq" class="section level2" number="2.2">
<h2><span class="header-section-number">2.2</span> MARS-seq</h2>
</div>
<div id="droplet-microfluidics-drop-seq" class="section level2" number="2.3">
<h2><span class="header-section-number">2.3</span> Droplet Microfluidics (DROP-seq)</h2>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:dropseq1"></span>
<img src="figures/DROP-seq.JPG" alt="Extraction and Processing of Single-Cell Transcriptomes by Drop-Seq [-@Macosko2015] " width="140%" />
<p class="caption">
FIGURE 2.2: Extraction and Processing of Single-Cell Transcriptomes by Drop-Seq <span class="citation">(<a href="#ref-Macosko2015" role="doc-biblioref">2015</a>)</span>
</p>
</div>
<p>DROP-seq enables high throughput profiling of thousands of individual cells in a
short period of time by encapsulating each cell individually with a unique
barcode in a droplet. The process takes place within a microfluidic device.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:dropseq2A"></span>
<img src="figures/DROP-seq2A.JPG" alt="Drop-Seq barcoding scheme" width="150%" />
<p class="caption">
FIGURE 2.3: Drop-Seq barcoding scheme
</p>
</div>
<p>After dissociating the complex tissue sample into individual cells, each cell is
suspended in droplets with microparticle beads which deliver barcoded primers
within a microfluidic device. When cells within the droplet are lysed, the mRNA
of these cells is attached to microparticles.A set of beads are generated with
the reverse transcription of mRNAs into cDNAs. These sets of beads are called
<code>STAMPs</code>, “single-cell transcriptomes attached to microparticles.” Finally, an
RNA-seq library with about 10,000 single-cell transcriptomes was constructed by
barcode <code>STAMPs</code> amplifying.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:dropseq2B"></span>
<img src="figures/DROP-seq2B.JPG" alt="Sequence of primers on the microparticle" width="150%" />
<p class="caption">
FIGURE 2.4: Sequence of primers on the microparticle
</p>
</div>
<p>Each barcoded primer bead contains an unique cell barcode and, an unique
molecular identifiers <code>(UMI)s</code> along with the common PCR handle. The common PCR
handle sequences used for PCR amplification after <code>STAMPs</code> formation. The cell
barcodes are unique for each droplets, therefore, each microparticles contains
primers with the same cell barcode. <code>UMIs</code> are unique barcodes to each
transcript, <code>UMIs</code> are counted digitially to count unique number of mRNA
transcripts. And, counting <code>UMIs</code> also removes the uncertainty of PCR
duplicates from transcript copy numbers.</p>
<p><em>Limitations of DROP-seq: </em></p>
<ul>
<li>Most of the droplets are empty and, very few of the droplets contains both a
cell and microparticle. Thus, it is not an efficient method.</li>
</ul>
</div>
<div id="x-genomics-chromium" class="section level2" number="2.4">
<h2><span class="header-section-number">2.4</span> 10X Genomics Chromium</h2>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:Genomicsx10"></span>
<img src="figures/10XGenomics.jpg" alt="10x Genomics Chromium scheme [-@Zheng2017]" width="150%" />
<p class="caption">
FIGURE 2.5: 10x Genomics Chromium scheme <span class="citation">(<a href="#ref-Zheng2017" role="doc-biblioref">2017</a>)</span>
</p>
</div>
<p>The 10X Genomics method is very similar to DROP-seq and, it is first described
in <span class="citation">(<a href="#ref-Zheng2017" role="doc-biblioref">2017</a>)</span>. First, the barcoded beads are mixed with the cell suspension.
Then, the mixture is capsulated with the emulsion generated by oil. The
microfluidic device and the barcoded beads of the 10X Genomics are engineered to
increase the efficiency of barcoding cells. Therefore almost 90% of the droplets
contain a cell and a barcoded bead. The main difference between the 10X Genomics
Chromium and DROP-seq is the reverse transcription occurs within the droplets.
After the amplification process, the transcriptomics library is constructed. For
each lanes of a 10X Genomics Chromium chip, approximately 20,000 cells can be
barcoded and sequenced simultaneously.</p>
<p><em>Advantages of 10X Genomics Chromium:</em></p>
<ul>
<li>10X Genomics is a commercial platform</li>
<li>Data obtained is high-throughput</li>
<li>Almost 10,000s reads per cell</li>
<li>Fast, inexpensive</li>
<li>3’ transcript sequencing</li>
<li>Single-nucleus sequencing possible
<ul>
<li>Less bias by cell morphology</li>
<li>Only captures nuclear RNA</li>
</ul></li>
</ul>
</div>
<div id="nanowell-platforms" class="section level2" number="2.5">
<h2><span class="header-section-number">2.5</span> Nanowell Platforms</h2>
</div>
<div id="sample-indexing-illumina" class="section level2" number="2.6">
<h2><span class="header-section-number">2.6</span> Sample Indexing (Illumina)</h2>
</div>
<div id="single-cell-combinatorial-indexing-scirna-seq" class="section level2" number="2.7">
<h2><span class="header-section-number">2.7</span> Single-cell Combinatorial Indexing: sciRNA-seq</h2>

</div>
</div>
<h3>References</h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Macosko2015" class="csl-entry">
Macosko, Evan Z., Anindita Basu, Rahul Satija, James Nemesh, Karthik Shekhar, Melissa Goldman, Itay Tirosh, et al. 2015. <span>“Highly Parallel Genome-Wide Expression Profiling of Individual Cells Using Nanoliter Droplets.”</span> <em>Cell</em> 161 (5): 1202–14. https://doi.org/<a href="https://doi.org/10.1016/j.cell.2015.05.002">https://doi.org/10.1016/j.cell.2015.05.002</a>.
</div>
<div id="ref-Svensson2018" class="csl-entry">
Svensson, Valentine, Roser Vento-Tormo, and Sarah A. Teichmann. 2018. <span>“<span class="nocase">Exponential scaling of single-cell RNA-seq in the past decade</span>.”</span> Nature Publishing Group. <a href="https://doi.org/10.1038/nprot.2017.149">https://doi.org/10.1038/nprot.2017.149</a>.
</div>
<div id="ref-Zheng2017" class="csl-entry">
Zheng, Grace, Jessica Terry, Phillip Belgrader, Paul Ryvkin, Zachary Bent, Ryan Wilson, Solongo Ziraldo, et al. 2017. <span>“Massively Parallel Digital Transcriptional Profiling of Single Cells.”</span> <em>Nature Communications</em> 8 (January): 14049. <a href="https://doi.org/10.1038/ncomms14049">https://doi.org/10.1038/ncomms14049</a>.
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="why-single-cell-analysis.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="single-cell-rna-seq-data-processing-and-quality-control.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/igunduz/the-book-of-a-cell/edit/master/02-scRNAseq-Technologies.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": "https://github.com/igunduz/the-book-of-a-cell/blob/master/02-scRNAseq-Technologies.Rmd",
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
